### ISSUE TITLE
Can't use Field Filter on foreign key of type/UUID

### ISSUE THREAD

**Issue by SandyRogers** [metabase/metabase#7955]
- Your browser and the version: MetaBase for Mac 29.3.0
- Your operating system: MacOS 10.13.5
- Your databases: Postgres 
- Metabase version: 29.3.0
- Metabase hosting environment: Mac app
- Metabase internal database: H2 (default)

Start with a Postgres database with two tables:

A table called `company` with this schema:

| COLUMN DATA | TYPE |
| ------------|-----------|
| id | type/UUID |
| name | type/Text |

A table called `team` with this schema:

| COLUMN DATA | TYPE |
| ------------|-----------|
|company_id | type/UUID|
|id | type/UUID|
|name | type/Text|

`team.company_id` is a foreign key to `company.id`.

I construct a query like:
`select * from team`
and then add a filter by company,
`select * from team where company_id=uuid({{company}})`,
where `{{company}}` is a basic text variable in the query builder.

As you see it is necessary to cast the UUID entered into the variable widget to UUID type, otherwise:
```
org.postgresql.util.PSQLException: ERROR: operator does not exist: uuid = character varying Hint: No operator matches the given name and argument type(s). You might need to add explicit type casts. Position: 159
```

This is do-able when using a basic text variable, but if I switch to:
`select * from team_team where {{company}}`,
and setup `{{company}}` as a Field Filter pointing at `team.company_id`, I then get the above error again.

(In the Data Model, `team.company_id` field is set as type: Foreign Key > Company → ID. With a Search Box filtering type. This should be supported as a filter field since it is of type foreign key.)

It looks like there is some code to decide whether or not to cast an id to numeric type or not, but not to cast to UUID type. E.g. [PR 7232](https://github.com/metabase/metabase/pull/7232/files)

**Comment by coderberry** [comment]
I am also having this issue

**Comment by gszep** [comment]
fyi : the above fix does not work for the case
```
select * from team_team
where ...
and {{company}}
```
but it should

**Comment by alexpusch** [comment]
I second @gszep replay. This issue should be reopened

**Comment by flamber** [comment]
Reopening based on comments.
@gszep and @alexpusch are you using the latest release (0.34.2) and seeing the same error as the original post `ERROR: operator does not exist: uuid = character varying` ?

**Comment by alexpusch** [comment]
@flamber Yes, 0.34.2, postgres 11
table has a uuid field, I defined a filter (`where {{fieldName}}`) of type 'field filter' that is mapped to the uuid field and set widget type 'category'

**Comment by walterl** [comment]
Fixed in #12083 (thanks to @MxKaras).
